### PR TITLE
bt-10: fix radio component

### LINKED ISSUE #163: [BUG] Can't select value in radiogroup component
**Description:**
The radio item is clicked but value isn't chosen.

**Steps to reproduce:**

Create RadioGroup element

**Expected Result:**

The value is clicked and chosen

**Severity:**
Blocker (S1)

### PR DESCRIPTION
**Pull request by savinaDaria**
-  changed component structure
- added hooks for form controller
image
image

### CODE REVIEW

**Review by JosuerBague — APPROVED** [review]
Working radio buttons now.

File: frontend/src/bundles/common/components/radio/radio-group.tsx [hunk 1]
```diff
@@ -1,50 +1,90 @@
...
+    type PathValue,
+} from 'react-hook-form';
 
-import { Radio } from './radio-item.js';
+import {
+    useCallback,
+    useFormController,
+} from '~/bundles/common/hooks/hooks.js';
+
+import { getValidClassNames } from '../../helpers/helpers.js';
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common/components/radio/radio-group.tsx]:
```suggestion
import { getValidClassNames } from '~/bundles/common/helpers/helpers.js';
```

File: frontend/src/bundles/common/components/radio/radio-group.tsx [hunk 2]
```diff
@@ -1,50 +1,90 @@
...
+    isDisabled = false,
+    row = true,
+    className = '',
+}: Properties<T>): JSX.Element => {
+    const radioGroupClasses = getValidClassNames(
+        styles['radio-group'],
+        className,
+    );
+    const firstElementIndex = 0;
+    const firstOption = options[firstElementIndex].value;
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common/components/radio/radio-group.tsx]:
```suggestion
    const [{ value: firstOption }] = options;
```
```suggestion
    const [{ value: firstOption }] = options;
```

File: frontend/src/bundles/common/components/radio/radio-group.tsx [hunk 3]
```diff
@@ -1,50 +1,90 @@
...
 };
 
-type Properties = {
+type Properties<T extends FieldValues> = {
+    control: Control<T, null>;
+    name: FieldPath<T>;
     options: Option[];
-    value: string;
-    className?: string;
     row?: boolean;
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common/components/radio/radio-group.tsx]: Boolean variable should have prefix has..., is... etc

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/radio/radio-group.tsx [hunk 4]
```diff
@@ -1,50 +1,89 @@
-import { FormControlLabel, RadioGroup as MuiRadioGroup } from '@mui/material';
+import {
+    FormControlLabel,
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/radio/radio-group.tsx]: looks like we have this - https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/blob/2f3ff530a4486c3603a9ca616b747934bc054b62/frontend/src/bundles/common/components/components.ts#L6

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/radio/radio-group.tsx (hunk 4 above)
Comment (savinaDaria) [reply frontend/src/bundles/common/components/radio/radio-group.tsx]: @nikita-remeslov  our custom FormLabel doesn't use FormControlLabel from mui, it uses FormLabel

**Review by JosuerBague — APPROVED** [review]
I merged this branch into a copy test branch of bt-115 (sign-up form) where I am using the radio group component.

It works. I can set default values and form validates correctly.

Removes the need for using `Controller` in my branch.

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**